lib/pkcs7.js

```diff
@@ -322,7 +323,7 @@ p7.createSignedData = function() {
         signatureAlgorithm: forge.pki.oids.rsaEncryption,
         signature: null,
         authenticatedAttributes: authenticatedAttributes,
-        unauthenticatedAttributes: []
+        unauthenticatedAttributes: signer.unauthenticatedAttributes || []
```

Comment:
Does `unauthenticatedAttributes` need similar checks as above for `authenticatedAttributes`?

Reply:
I am not sure, in `authenticatedAttributes` is at least one messageDigest attribute required... I assume the same condition won't be required for `unauthenticatedAttributes`, because messageDigest is already in `authenticatedAttributes`.